File: bubble_sort/bubble_sort.rb
require_relative '../lib/qtool'

class BubbleSort
  include Qtool::DataPreview

  def initialize(data)
    @data = data
  end

  def sort
    size = @data.length

    for i in 0..size-1
      for j in 0..size-2
        if @data[j] > @data[j+1]
          @data[j], @data[j+1] = @data[j+1] , @data[j]
        end
      end
    end

    puts "::BubbleSort::"
    preview @data
  end

end

data = Qtool::DataCreator.create(10,100)
buble_sort = BubbleSort.new(data)
buble_sort.preview(data)
buble_sort.sort

File: bubble_sort/bubble_sort.js
var qtool = require("../lib/qtool");

function _bubbleSort(data) {
  this.data = data;
  this.preview = function(data) {
    qtool.dataPreview.preview(data);
  };
  this.sort = function() {
    var length = this.data.length;
    for(var i = 1; i < length; i++) {
      for(var j = 0; j < length - 1; j++) {
        var temp = this.data[j];
        if(this.data[j] > this.data[j+1]){
          this.data[j] = this.data[j+1];
          this.data[j+1] = temp;
        };
      };
    };
    console.log("::BubbleSort::");
    qtool.dataPreview.preview(this.data);
  };
};

var data = qtool.dataCreator.create(10,100);
var bubleSort = new _bubbleSort(data);
bubleSort.preview(data);
bubleSort.sort();
